File: go/datastructure/teststringbuffer_test.go
package datastructure

import (
    "testing"
    lang "github.com/otterop/otterop/go/lang"
    test "github.com/otterop/otterop/go/test"
)


func testStringBufferNew() *TestStringBuffer {
    this := new(TestStringBuffer)
    this.TestBase = test.TestBaseNew()
    return this
}

type TestStringBuffer struct {
    *test.TestBase
}




func (this *TestStringBuffer) empty()  {
    var sb *StringBuffer = StringBufferNew()
    var s *lang.String = sb.OOPString()
    this.AssertTrue(s.CompareTo(lang.StringWrap(lang.StringLiteral(""))) == 0, lang.StringWrap(lang.StringLiteral("Should be an empty string")))
}

func (this *TestStringBuffer) addMoreStrings()  {
    var sb *StringBuffer = StringBufferNew()
    sb.Add(lang.StringWrap(lang.StringLiteral("a")))
    var s *lang.String = sb.OOPString()
    this.AssertTrue(s.CompareTo(lang.StringWrap(lang.StringLiteral("a"))) == 0, lang.StringWrap(lang.StringLiteral("Should be equals to 'a'")))
    sb.Add(lang.StringWrap(lang.StringLiteral(",b")))
    s = sb.OOPString()
    this.AssertTrue(s.CompareTo(lang.StringWrap(lang.StringLiteral("a,b"))) == 0, lang.StringWrap(lang.StringLiteral("Should be equals to 'a,b'")))
}

func TestTestStringBuffer(t *testing.T) {
    t.Run("empty", func(t *testing.T) {
        test := testStringBufferNew()
        test.SetGoTestingT(t)
        test.empty()
    })
    t.Run("addMoreStrings", func(t *testing.T) {
        test := testStringBufferNew()
        test.SetGoTestingT(t)
        test.addMoreStrings()
    })
}

File: ts/datastructure/otterop/datastructure/TestStringBuffer.test.ts
import { String } from '@otterop/lang/String';
import { TestBase } from '@otterop/test/TestBase';
import { StringBuffer } from './StringBuffer';

export class TestStringBuffer extends TestBase {

    public empty() : void {
        let sb : StringBuffer = new StringBuffer();
        let s : String = sb.OOPString();
        this.assertTrue(s.compareTo(String.wrap("")) == 0, String.wrap("Should be an empty string"));
    }

    public addMoreStrings() : void {
        let sb : StringBuffer = new StringBuffer();
        sb.add(String.wrap("a"));
        let s : String = sb.OOPString();
        this.assertTrue(s.compareTo(String.wrap("a")) == 0, String.wrap("Should be equals to 'a'"));
        sb.add(String.wrap(",b"));
        s = sb.OOPString();
        this.assertTrue(s.compareTo(String.wrap("a,b")) == 0, String.wrap("Should be equals to 'a,b'"));
    }
}


test('empty', () => {
    new TestStringBuffer().empty();
});

test('addMoreStrings', () => {
    new TestStringBuffer().addMoreStrings();
});
